src/framed_write.rs

```diff
@@ -124,7 +124,9 @@ where
         Poll::Ready(Ok(()))
     }
     fn poll_close(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Result<(), Self::Error>> {
-        Pin::new(&mut self.inner).poll_close(cx).map_err(Into::into)
+        let this = &mut *self;
+        ready!(Pin::new(&mut this.inner).poll_flush(cx).map_err(Into::into))?;
+        Pin::new(&mut this.inner).poll_close(cx).map_err(Into::into)
```

Comment:
I'm not an expert, but this is just a wrapper on the inner type. So I think you would flush the inner type when `FramedWrite2` get's flushed, and you would not add a flush in poll_close. Maybe it's a good idea to flush `self` before closing, since it has a buffer, but I don't know if that's how it's meant to be done. I definitely think someone more knowledgeable about the Sink trait should look at that or could at least answer questions.

Reply:
See 5396b4c6914bdec8c74d521e77de23cbdf118d1c.